pkg/operator/ceph/object/config.go

```diff
@@ -215,12 +216,39 @@ func (c *clusterConfig) generateMonConfigOptions(rgwConfig *rgwConfig) (map[stri
 	for flag, val := range c.store.Spec.Gateway.RgwConfig {
 		if currVal, ok := configOptions[flag]; ok {
 			// RGW might break with some user-specified config overrides; log clearly to help triage
-			logger.Infof("overriding object store %q RGW config option %q (current value %q) with user-specified rgwConfig %q",
+			logger.Infof("overriding object store %q RGW config option %q with user-specified rgwConfig",
+				fmt.Sprintf("%s/%s", c.store.Namespace, c.store.Name), flag)
+			logger.Debugf("overriding object store %q RGW config option %q (current value %q) with user-specified rgwConfig %q",
 				fmt.Sprintf("%s/%s", c.store.Namespace, c.store.Name), flag, currVal, val)
 		}
 		configOptions[flag] = val
 	}
 
+	for flag, val := range c.store.Spec.Gateway.RgwConfigFromSecret {
+		if val.Key == "" || val.Name == "" {
+			logger.Warningf("invalid object store %q rgwConfigFromSecret value %q=%+v", fmt.Sprintf("%s/%s", c.store.Namespace, c.store.Name), flag, val)
+			continue
+		}
+		secret, err := c.context.Clientset.CoreV1().Secrets(c.clusterInfo.Namespace).Get(c.clusterInfo.Context, val.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("unable to lookup secret for object store %q rgwConfigFromSecret %q=%+v", fmt.Sprintf("%s/%s", c.store.Namespace, c.store.Name), flag, val)
+		}
+		secretVal, ok := secret.Data[val.Key]
+		if !ok {
+			return nil, fmt.Errorf("unable to lookup secret for object store %q rgwConfigFromSecret %q: key %q not found in secret %q",
+				fmt.Sprintf("%s/%s", c.store.Namespace, c.store.Name), flag, val.Key, val.Name)
+		}
+
+		if currVal, ok := configOptions[flag]; ok {
+			// RGW might break with some user-specified config overrides; log clearly to help triage
+			logger.Infof("overriding object store %q RGW config option %q with user-specified rgwConfigFromSecret",
+				fmt.Sprintf("%s/%s", c.store.Namespace, c.store.Name), flag)
+			logger.Debugf("overriding object store %q RGW config option %q (current value %q) with user-specified rgwConfigFromSecret %q",
```

Comment:
This could expose a password, so let's use trace logging.

```suggestion
			logger.Tracef("overriding object store %q RGW config option %q (current value %q) with user-specified rgwConfigFromSecret %q",
```

Reply:
done